### PR TITLE
Team 5 sprint3

### LINKED ISSUE #269: Character: More player Attributes, enhancing player's interaction, UI
# **Description**
Enhance player interaction and engagement by introducing additional attributes and UI features. This includes implementing a scoring system, sound effects, new character interactions, modified key bindings, and achievements.

## **SubFeatures/components**
 1. **Add a scoring system:**  #296 
     - For each animal killed, a certain number of coins will be awarded to the player.
     - Display the accumulated coins in the UI.

2. **Spend Money**  
     - Players can trade coins for items from gambling (based on maps team's implementation of gamble room)

3. **Sound Effects for Damage**  
     - Create and implement different sound effects to enhance immersion when the player takes damage.

4. **New Character - Homeless Guy**  
   - Introduce a new character, "Homeless Guy," who interacts with the player.

5. **Modified Key Binding - Continuous Attack**  
     - When the attack key (e.g., mouse button or keyboard key) is long-pressed, the character will continuously shoot or attack until the key is released.

6. **Update UI**
   - Make UI similar to minecraft UI
   - Display scores
 
7. **Player Transformation into NPC** (collab with animals team)
  - Player can be transformed into NPC during the game 

8. **Speed for players**
  - Players have different speed attributes this time

# **Dependencies**
- Creation of Gamble Room.

# **Milestones**
- [x] #296 (Sep 30)
- [x] Spend Money (Sep 30)
- [x] Sound Effects for Damage (Sep 30)
- [x] New Character - Homeless Guy (Sep 30)
- [x] #286
- [x] Update UI (Sep 30)
- [x] Testing (Sep 30)
- [x] Documentation (Oct 2)

**Completion Deadline:** Oct 2

# **Extras (optional)**
1. **Achievements (Extra Task)**  
   - Implement an achievement system to track player progress and award them for completing specific challenges.
  

# **Documentation**
TBA

# **Members**
- @manya-k @SIN-Willett @VuKhanhVyHo @Jordamnmn @nadiadmello

### PR DESCRIPTION
**Pull request by manya-k**
# Description

- New player animations for build screen
- ability for players to hold one key down for continuous shoot
- scoring system based on animals' strength
- Saving and loading for player
- different sounds for player getting hit
- Inventory display is no won the bottom of the screen

NOTE: the left and right animation for different players will be fixed in sprint 4

Fixes / Closes #269 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Testing 
- [x] User Testing
- [x] Manual Testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by CameronBadman** [comment]
Fix Hobo, i feel cheated

File: source/core/src/main/com/csse3200/game/screens/MainGameScreen.java [hunk 1]
```diff
@@ -120,25 +121,38 @@ public MainGameScreen(GdxGame game) {
...
-                PLAYERS
-        ));
-        Entity player = playerFactory.createPlayer(
-                FileLoader.readClass(PlayerConfig.class, chosenPlayer).name);
+        this.playerFactory = new PlayerFactory(Arrays.stream(PLAYERS).toList());
+        Entity player;
+        if (shouldLoad) {
+            System.out.println("Should start from save file: {}" + shouldLoad);
+            player = playerFactory.createPlay("configs/player_save.json",
+                    gameOptions.difficulty);
```
Comment (IanP96) [inline source/core/src/main/com/csse3200/game/screens/MainGameScreen.java:137]: For sprint 4 we could have the difficulty saved as well maybe

File: source/core/src/main/com/csse3200/game/screens/MainGameScreen.java [hunk 2]
```diff
@@ -120,25 +121,38 @@ public MainGameScreen(GdxGame game) {
...
-        // todo confirm which players should be passed into PlayerFactory
-        this.playerFactory = new PlayerFactory(List.of(
-                PLAYERS
-        ));
-        Entity player = playerFactory.createPlayer(
-                FileLoader.readClass(PlayerConfig.class, chosenPlayer).name);
+        this.playerFactory = new PlayerFactory(Arrays.stream(PLAYERS).toList());
+        Entity player;
+        if (shouldLoad) {
+            System.out.println("Should start from save file: {}" + shouldLoad);
```
Comment (IanP96) [inline source/core/src/main/com/csse3200/game/screens/MainGameScreen.java:135]: This prints `Should start from save file: {}true`, also line 126 is basically the same message so I don't know why you need this

**Review by IanP96 — COMMENTED** [review]
looks good

**Comment by manya-k** [comment]
> Fix Hobo, i feel cheated

fixed!

File: source/core/assets/configs/player_2.json [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
+  "name": "Player 2",
   "health": 70,
-  "favouriteColour": "yellow",
-  "textureFilename": "images/player/player.png",
-  "textureAtlasFilename": "images/player/player.atlas"
+  "coins": 0,
+  "speed": {
+    "x": 3,
+    "y": 3
+  },
```
[the remark this answers is not in the document]
Comment (manya-k) [reply source/core/assets/configs/player_2.json]: done!

File: source/core/src/main/com/csse3200/game/screens/MainGameScreen.java (hunk 2 above)
Comment (manya-k) [reply source/core/src/main/com/csse3200/game/screens/MainGameScreen.java]: done

**Pull request closed without merging**